sia/src/rhp/rpc.rs

```diff
@@ -0,0 +1,613 @@
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCSettingsResponse {
+    pub settings: HostSettings,
+}
+
+/// RPCFormContractParams contains the parameters for forming a new contract.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFormContractParams {
+    pub renter_public_key: PublicKey,
+    pub renter_address: Address,
+    pub allowance: Currency,
+    pub collateral: Currency,
+    pub proof_height: u64,
+}
+
+/// RPCFormContractRequest is the request type for RPCFormContract.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFormContractRequest {
+    pub prices: HostPrices,
+    pub contract: RPCFormContractParams,
+    pub miner_fee: Currency,
+    pub basis: ChainIndex,
+    pub renter_inputs: Vec<SiacoinElement>,
+    pub renter_parents: Vec<Transaction>,
+}
+
+/// RPCFormContractResponse contains the host's Siacoin inputs for the contract
+/// formation transaction.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFormContractResponse {
+    pub host_inputs: Vec<SiacoinInput>,
+}
+
+/// RPCFormContractSecondResponse contains the renter's contract signature and
+/// Siacoin input signatures for the contract formation transaction.
+///
+/// At this point, the host has enough information to broadcast the formation.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFormContractSecondResponse {
+    pub renter_contract_signature: Signature,
+    pub renter_satisfied_policies: Vec<SatisfiedPolicy>,
+}
+
+/// RPCFormContractThirdResponse contains the finalized formation
+/// transaction set.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFormContractThirdResponse {
+    pub basis: ChainIndex,
+    pub transaction_set: Vec<Transaction>,
+}
+
+/// RPCRefreshContractParams contains the parameters for refreshing a contract.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCRefreshContractParams {
+    pub contract_id: FileContractID,
+    pub allowance: Currency,
+    pub collateral: Currency,
+}
+
+/// RPCRefreshContractRequest is the request type for RPCRefreshContract.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCRefreshContractRequest {
+    pub prices: HostPrices,
+    pub refresh: RPCRefreshContractParams,
+    pub miner_fee: Currency,
+    pub basis: ChainIndex,
+    pub renter_inputs: Vec<SiacoinElement>,
+    pub renter_parents: Vec<Transaction>,
+
+    pub challenge_signature: Signature,
```

Comment:
Is this alright to use from a DX perspective? Considering that you need to create the request first, then create the hash with a method and sign it and then assign the signature. As compared to having a constructor that does it.
e.g. I like how the account token solves it

Alternatively maybe we can have an unsigned type that gets consumed and creates a signed type when calling `sign` on it. Or some other more Rusty approach if there is one.

Same goes for the other types with challenge signatures.